### ISSUE TITLE
Fresh Generation fails : AssociationError for many-to-many self-associations

### ISSUE THREAD

**Issue by julienperrault** [sequelize/sequelize-auto#500]
Hi.
I'm on Sequelize 6.5 and sequelize-auto 0.7.11.

I have successfully generated models (cmd-line OR Programmatic API), but i can't use it (`initModels` fails)
I get an Association Error. 

```log
AssociationError [SequelizeAssociationError]: 'as' must be defined for many-to-many self-associations
    at new BelongsToMany
```

I looked around a bit and here's the kind of associations that come into conflict.

```js
  ElementA.belongsToMany(ElementA, { through: ElementAParent, foreignKey: "elementAId", otherKey: "parentId" });
  ElementA.belongsToMany(ElementB, { through: ElementBElementA, foreignKey: "elementAId", otherKey: "elementBId" });
```

<hr>

init-models.js

```js
var DataTypes = require("sequelize").DataTypes;
var _ElementAParent = require("./elementAParent");
var _ElementA = require("./elementA");
var _ElementBElementA = require("./elementBElementA");
var _ElementB = require("./elementB");

function initModels(sequelize) {

  var ElementA = _ElementA(sequelize, DataTypes);
  var ElementB = _ElementB(sequelize, DataTypes);

  ElementA.belongsToMany(ElementA, { through: ElementAParent, foreignKey: "elementAId", otherKey: "parentId" });
  ElementA.belongsToMany(ElementB, { through: ElementBElementA, foreignKey: "elementAId", otherKey: "elementBId" });

  return {
    ElementBElementA,
    ElementB,
    ElementAParent,
    ElementA,
  };
}
module.exports = initModels;
module.exports.initModels = initModels;
module.exports.default = initModels;

```

**Comment by steveschmitt** [comment]
Hmm.  It seems like it needs to be something like:
```js
ElementA.belongsToMany(ElementA, { as: "relatedA", through: ElementAParent, foreignKey: "elementAId", otherKey: "parentId" });
```

**Comment by steveschmitt** [comment]
Fixed in 0.8.0
